src/JNI/com/amazonaws/kinesis/video/producer/jni/KinesisVideoClientWrapper.cpp

```diff
@@ -55,31 +55,32 @@ KinesisVideoClientWrapper::KinesisVideoClientWrapper(JNIEnv* env,
 KinesisVideoClientWrapper::~KinesisVideoClientWrapper()
 {
     STATUS retStatus = STATUS_SUCCESS;
+    JNIEnv *env;
+
+    if (mJvm == NULL) {
+        return;
+    }
+
+    mJvm->GetEnv((PVOID*) &env, JNI_VERSION_1_6);
+
     if (IS_VALID_CLIENT_HANDLE(mClientHandle))
     {
         if (STATUS_FAILED(retStatus = freeKinesisVideoClient(&mClientHandle))) {
             DLOGE("Failed to free the producer client object");
```

Comment:
log the retstatus as well?

Reply:
`throwNativeException(env, EXCEPTION_NAME, "Failed to free the producer client object.", retStatus);` (below) will print the retStatus.